src/plugins/intel_gpu/tests/shape_infer/eltwise_si_test.cpp

```diff
@@ -0,0 +1,118 @@
+    auto eltwise_prim = std::make_shared<eltwise>("output", "input1", "input2", p.mode, p.auto_broadcast_spec);
+
+    cldnn::program prog(engine);
+
+    auto& input1_node = prog.get_or_create(input1_prim);
+    auto& input2_node = prog.get_or_create(input2_prim);
+    auto& eltwise_node = prog.get_or_create(eltwise_prim);
+    program_wrapper::add_connection(prog, input1_node, eltwise_node);
+    program_wrapper::add_connection(prog, input2_node, eltwise_node);
+    auto res = eltwise_inst::calc_output_layouts<ov::PartialShape>(eltwise_node, *eltwise_node.get_kernel_impl_params());
+
+    auto expected_layout = layout{p.expected_shape, data_types::f32, format::bfyx};
+    ASSERT_EQ(res.size(), 1);
+    ASSERT_EQ(res[0], expected_layout);
+}
+
+
+
+INSTANTIATE_TEST_SUITE_P(smoke, eltwise_si_test,
+    testing::ValuesIn(std::vector<eltwise_si_test_params>{
+        {
+            ov::PartialShape{2, 1, 5},
+            ov::PartialShape{1, 4, 1},
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY),
+            ov::PartialShape{2, 4, 5}
+        },
+        {
+            ov::PartialShape{1, 1, 5},
+            ov::PartialShape{5, 2, 1, 3},
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY),
+            ov::PartialShape{5, 2, 5, 3}
+        },
+        {
+            ov::PartialShape{2, 3, 4, 5},
+            ov::PartialShape{4, 5},
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::PDPD, -1),
+            ov::PartialShape{2, 3, 4, 5}
+        },
+        {
+            ov::PartialShape{2, 3, 4, 5},
+            ov::PartialShape{1, 3},
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::PDPD),
+            ov::PartialShape{2, 3, 4, 5}
+        },
+        {
+            ov::PartialShape{2, -1, 5},
+            ov::PartialShape{1, 4, 1},
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY),
+            ov::PartialShape{2, 4, 5}
+        },
+        {
+            ov::PartialShape::dynamic(3),
+            ov::PartialShape{1, 4, 1},
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY),
+            ov::PartialShape{-1, 4, -1}
+        },
+        {
+            ov::PartialShape{2, 1, 5},
+            ov::PartialShape::dynamic(3),
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY),
+            ov::PartialShape{2, -1, 5}
+        },
+        {
+            ov::PartialShape{1,   1, 512,  1},
+            ov::PartialShape{-1, -1, 1024, 512},
+            cldnn::eltwise_mode::sum,
+            ov::op::AutoBroadcastSpec(ov::op::AutoBroadcastType::NUMPY),
+            ov::PartialShape::dynamic(4)
```

Comment:
Shape infer for cases when 1 input is constant is not covered. 
Also, since we have non-trivial out format/dt selection logic, I'd expect some tests with correctness check for those parameters too. So maybe tests parameter should include fully specified layout for inputs and expected outputs.
And one more thing is 1d tensors (both const and non-const) - need to check such kind of broadcast as well.

Reply:
I added more test params including const / non-const 1d tensors and logical/comparison operation output data type.
Please review it again. thanks.